webapp/channels/src/components/sidebar/unread_channels.tsx

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+import {useIntl} from 'react-intl';
+
+import {SidebarCategoryHeaderStatic} from './sidebar_category_header';
+import SidebarChannel from './sidebar_channel';
+import SidebarCategoryGenericMenu from './sidebar_category/sidebar_category_menu/sidebar_category_generic_menu';
+import MarkAsReadMenuItem from './sidebar_category/sidebar_category_menu/mark_as_read_menu_item';
+import * as Menu from 'components/menu';
+import CreateNewCategoryMenuItem from './sidebar_category/sidebar_category_menu/create_new_category_menu_item';
+import {trackEvent} from 'actions/telemetry_actions';
+import {useDispatch, useSelector} from 'react-redux';
+import {getUnreadChannels} from 'selectors/views/channel_sidebar';
+import {viewMultipleChannels} from 'mattermost-redux/actions/channels';
+
+type Props = {
+    setChannelRef: (channelId: string, ref: HTMLLIElement) => void;
+};
+
+export default function UnreadChannels({
+    setChannelRef,
+}: Props) {
+    const intl = useIntl();
+    const unreadChannels = useSelector(getUnreadChannels);
```

Comment:
There are some changes on this file, like using the redux hooks, but it is mostly a file move that has not been properly recognized by git.